app/api/orders.py

```diff
@@ -319,7 +317,7 @@ def before_update_object(self, order, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-        if data.get('amount') or data.get('is_billing_enabled'):
+        if data.get('amount') and (data.get('is_billing_enabled') or order.event.is_billing_info_mandatory) :
```

Comment:
This will again, require billing info even if the organizer has asked not to. Please read the issue again. I don't think you understand what the goal is.

Can you tell me what is the goal of this PR?

Reply:
This PR goal is, If a order is paid and organiser has made billing mandatory , user has to fill it.
And if its not mandatory then user has option to fill or not

Reply:
Also Billing information is only needed for paid order hence a condition for `data.get(amount)`

Reply:
> what

If organiser has not asked for billing info, Then it wont require as I have used `and` over there

Reply:
@iamareebjamal

Reply:
Yeah, and this is going to require billing info even when organizer has set `event.is_billing_info_mandatory` = False

Reply:
> Yeah, and this is going to require billing info even when organizer has set `event.is_billing_info_mandatory` = False

It will billing info only then when the user wants to enter it